arch/x86/timing.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arch/x86/arch.h>
+#include <kernel.h>
+#include <sys_clock.h>
+#include <timing/timing.h>
+
+static uint64_t tsc_freq;
+
+void timing_x86_init(void)
+{
+	uint32_t cyc_start = k_cycle_get_32();
+	uint64_t tsc_start = z_tsc_read();
+
+	k_busy_wait(10 * USEC_PER_MSEC);
```

Comment:
seems a fairly short timing loop, does this produce stable/accurate values on, say, and upsquared?

Reply:
Seems stable to me. The differences between different runs are in the kHz level.